### ISSUE TITLE
[Feature Request] Security Station: Moving protection field

### ISSUE THREAD

**Issue by nirakoji** [TeamPneumatic/pnc-repressurized#939]
I mentioned this briefly on Discord the other day, documenting it here in a bit more detail:

In short, I'm thinking of a feature that allows us to move the protection cube of the security station to allow the station to be off-center for both functional and aesthetic reasons.

I'm thinking of an XYZ offset ability, clamped by Range Upgrades and stock range, perhaps range-1, so that the security station is always within it's own protection range.

Perhaps as part of that, we could have an option to define the coordinates of it for a non-square cuboid, with the max parameters still clamped by the range upgrades. This would make sharing spaces/security levels a bit more flexible and dynamic.

The above changes would also be very helpful in keeping "teams" integrated within PNC via permissions on security stations. I've had mixed experiences with other team mods/features on servers I've adminned and modpacks I've curated.

**Comment by desht** [comment]
Non-cuboid shaped areas could conceivably be handled with a GPS Area tool; in fact, that would also address the offsetting requirement.  Would need care to ensure performance doesn't suffer, though (it's a more heavyweight solution than just check if a blockpos is within a certain distance from the Security Station).
